### PR TITLE
Add Bithumb Spot exchange

### PR DESCRIPTION
**Pull request by artememelin**
### Pull request checklist

- [x] Did you bump the project version?
- [x] Did you add a description to the Pull Request?
- [ ] Did you add new tests?
- [x] Did you add reviewers?

### CODE REVIEW

**Comment by artememelin** [comment]
:blue_book: A preview of the documentation will be [here](https://bhftbootcamp.github.io/CryptoAPIs.jl/previews/PR35/) soon

File: src/Bithumb/Spot/API/Candle.jl [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+    x == h6  && return "6h"
+    x == h12 && return "12h"
+    x == h24  && return "24h"
+end
+
+Serde.SerQuery.ser_ignore_field(::Type{CandleQuery}, ::Val{:order_currency}) = true
+
+Serde.SerQuery.ser_ignore_field(::Type{CandleQuery}, ::Val{:payment_currency}) = true
+
+Serde.SerQuery.ser_ignore_field(::Type{CandleQuery}, ::Val{:interval}) = true
```
Comment (yshadrin) [inline src/Bithumb/Spot/API/Candle.jl:38]:
```suggestion
Serde.SerQuery.ser_ignore_field(::Type{CandleQuery}, ::Val{:order_currency}) = true
Serde.SerQuery.ser_ignore_field(::Type{CandleQuery}, ::Val{:payment_currency}) = true
Serde.SerQuery.ser_ignore_field(::Type{CandleQuery}, ::Val{:interval}) = true
```
remove line breaks between methods

File: src/Bithumb/Spot/API/OrderBook.jl [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+
+struct OrderBookData <: BithumbData
+    asks::Vector{Level}
+    bids::Vector{Level}
+    order_currency::String
+    payment_currency::String
+    timestamp::NanoDate
+end
+
+function CryptoAPIs.prepare_json!(::Type{T}, json::Dict{String,Any}) where {T<:Data{Dict{String,OrderBookData}}}
```
Comment (yshadrin) [inline src/Bithumb/Spot/API/OrderBook.jl:38]:
```suggestion
function prepare_json!(::Type{T}, json::Dict{String,Any}) where {T<:Data{Dict{String,OrderBookData}}}
```

File: src/Bithumb/Spot/API/OrderBook.jl [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+    OrderBookData,
+    order_book
+
+using Serde
+using Dates, NanoDates, TimeZones
+
+using CryptoAPIs
+using CryptoAPIs.Bithumb
+using CryptoAPIs.Bithumb: Data
+using CryptoAPIs: Maybe, APIsRequest
```
Comment (yshadrin) [inline src/Bithumb/Spot/API/OrderBook.jl:13]:
```suggestion
using CryptoAPIs.Bithumb
using CryptoAPIs.Bithumb: Data
using CryptoAPIs: Maybe, APIsRequest
import CryptoAPIs: prepare_json!
```
I suggest we do this for uniformity

File: src/Bithumb/Spot/API/UserTransactionsLog.jl [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+to_pretty_json(result.result)
+```
+
+## Result:
+
+```json
+
+```
+"""
+function user_transactions_log(client::BithumbClient, query::UserTransactionsLogQuery)
```
Comment (yshadrin) [inline src/Bithumb/Spot/API/UserTransactionsLog.jl:106]: I suggest for this endpoint calling the method `user_transactions`

Comment (yshadrin) [reply src/Bithumb/Spot/API/UserTransactionsLog.jl:106]: and the corresponding renaming of structures and file names

File: src/Bithumb/Spot/API/UserTransactions.jl [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+
+```json
+
+```
+"""
+function user_transactions(client::BithumbClient, query::UserTransactionsQuery)
+    return APIsRequest{Data{Vector{UserTransactionsData}}}("POST", "info/user_transactions", query)(client)
+end
+
+function user_transactions(client::BithumbClient = Bithumb.Spot.public_client; kw...)
```
Comment (yshadrin) [inline src/Bithumb/Spot/API/UserTransactions.jl:112]:
```suggestion
function user_transactions(client::BithumbClient; kw...)
```
method is private

File: src/Bithumb/Utils.jl [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+    return unixnanos2nanodate(parse(Int64, x) * 1e6)
+end
+
+function Serde.deser(::Type{Data}, ::Type{<:Maybe{NanoDate}}, x::AbstractString)::NanoDate
+    return unixnanos2nanodate(parse(Int64, x) * 1e6)
+end
+
+function Serde.deser(::Type{Data}, ::Type{<:Maybe{NanoDate}}, x::Int64)::NanoDate
+    return unixnanos2nanodate(parse(Int64, x) * 1e6)
+end
```
Comment (dmitrii-doronin) [inline src/Bithumb/Utils.jl:17]: Does this work?

**Review by yshadrin — APPROVED** [review]

**Review by gryumov — APPROVED** [review]

**Pull request merged**